tests/integration/features/servicemesh_feature_test.go

```diff
@@ -43,136 +45,227 @@ var _ = Describe("Service Mesh feature", func() {
 		Expect(err).ToNot(HaveOccurred())
 	})
 
-	AfterEach(func() {
-
-	})
-
 	Describe("preconditions", func() {
 
-		When("operator is not installed", func() {
+		Context("operator setup", func() {
 
-			It("should fail using precondition check", func() {
-				// given
-				featuresHandler := feature.ClusterFeaturesHandler(dsci, func(handler *feature.FeaturesHandler) error {
-					verificationFeatureErr := feature.CreateFeature("no-service-mesh-operator-check").
-						For(handler).
-						UsingConfig(envTest.Config).
-						PreConditions(servicemesh.EnsureServiceMeshOperatorInstalled).
-						Load()
+			When("operator is not installed", func() {
 
-					Expect(verificationFeatureErr).ToNot(HaveOccurred())
+				It("should fail using precondition check", func() {
+					// given
+					featuresHandler := feature.ClusterFeaturesHandler(dsci, func(handler *feature.FeaturesHandler) error {
+						verificationFeatureErr := feature.CreateFeature("no-service-mesh-operator-check").
+							For(handler).
+							UsingConfig(envTest.Config).
+							PreConditions(servicemesh.EnsureServiceMeshOperatorInstalled).
+							Load()
 
-					return nil
-				})
+						Expect(verificationFeatureErr).ToNot(HaveOccurred())
+
+						return nil
+					})
 
-				// when
-				applyErr := featuresHandler.Apply()
+					// when
+					applyErr := featuresHandler.Apply()
 
-				// then
-				Expect(applyErr).To(MatchError(ContainSubstring("failed to find the pre-requisite operator subscription \"servicemeshoperator\"")))
+					// then
+					Expect(applyErr).To(MatchError(ContainSubstring("failed to find the pre-requisite operator subscription \"servicemeshoperator\"")))
+				})
 			})
-		})
 
-		When("operator is installed", func() {
-			var smcpCrdObj *apiextensionsv1.CustomResourceDefinition
+			When("operator is installed", func() {
+				var smcpCrdObj *apiextensionsv1.CustomResourceDefinition
 
-			BeforeEach(func() {
-				err := fixtures.CreateSubscription(fixtures.OssmSubscription, "openshift-operators", envTestClient)
-				Expect(err).ToNot(HaveOccurred())
+				BeforeEach(func() {
+					err := fixtures.CreateSubscription(fixtures.OssmSubscription, "openshift-operators", envTestClient)
```

Comment:
As tests run in sequence there is no need now to remove subscription, but I will clean this up in the follow-up PR.